### PR TITLE
[Admin] connect flash messages to the `ui/toast` component

### PR DESCRIPTION
**Pull request by elia**
## Summary

- If multiple messages are present they'll stack on top of each other.
- Based on #5236, please review just the last commit

image

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

**Review by waiting-for-dev — COMMENTED** [review]
Great work defining the different responsibilities we need :muscle: I left a suggestion. I am interested in your thoughts.

File: admin/app/views/layouts/solidus_admin/application.html.erb [hunk 1]
```diff
@@ -4,19 +4,28 @@
...
 
-    <main class="
-      col-start-2 col-end-4
-      lg:col-start-3 lg:col-end-12
-      py-6 px-4
+  <body>
+    <div class="fixed right-3 bottom-3 flex flex-col gap-3">
+      <% flash.each do |key, message| %>
+        <%= render component("ui/toast").new(text: message, scheme: key == :error ? :error : :default) %>
+      <% end %>
```
[the remark this answers is not in the document]
Comment (kennyadsl) [reply admin/app/views/layouts/solidus_admin/application.html.erb:12]: 👍 Let me also add that at some point we'll need to differentiate between toast and "normal" flash messages, so there will be some logic involved. I'm not yet sure where that logic needs to be determined but having a component for all the messages sounds like a good idea.

**Comment by waiting-for-dev** [comment]
It would be worth it to rebase on top of `nebulab/admin`, as we now have automatic a11y checks, and something could break after the merge.

File: admin/app/views/layouts/solidus_admin/application.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (elia) [reply admin/app/views/layouts/solidus_admin/application.html.erb:12]: Yeah, the reason I didn't go too deep with this is exactly that we still need to figure out how we'll use them. Right now we barely have any flash message (the first ones will be added by performing batch actions).

I think eventually we'll also need some sort of layout component that can take care of it and possibly move the grid classes out of the sidebar, but I think that's a separate task, also it's possible we'll need to show toast messages from JS. But it all seems premature at this point of the development.

What do you think about waiting until we need something more sophisticated and handling it when it hurts?

**Review by waiting-for-dev — APPROVED** [review]

File: admin/app/views/layouts/solidus_admin/application.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kennyadsl) [reply admin/app/views/layouts/solidus_admin/application.html.erb:12]: Works for me, moving that row erb into a component seems ok for a future PR.

**Review by kennyadsl — APPROVED** [review]

**Pull request merged**